src/components/SelectionList/BaseSelectionList.tsx

```diff
@@ -365,6 +368,40 @@ function BaseSelectionList<TItem extends ListItem>(
         updateAndScrollToFocusedIndex(newSelectedIndex);
     }, [canSelectMultiple, flattenedSections.allOptions.length, prevTextInputValue, textInputValue, updateAndScrollToFocusedIndex]);
 
+    useEffect(
+        () => () => {
+            if (!itemFocusTimeoutRef.current) {
+                return;
+            }
+            clearTimeout(itemFocusTimeoutRef.current);
+        },
+        [],
+    );
+
+    const scrollAndHighlightItem = useCallback(
```

Comment:
Could you add a plain english explanation to a doc comment of what is being done in this method?